dataset/src/androidMain/kotlin/org/jetbrains/kotlinx/dl/dataset/preprocessing/camerax/ImageproxyUtlls.kt

```diff
@@ -0,0 +1,187 @@
+package org.jetbrains.kotlinx.dl.dataset.preprocessing.camerax
+
+import android.graphics.*
+import androidx.camera.core.ImageProxy
+import org.jetbrains.kotlinx.dl.dataset.preprocessing.bitmap.Rotate
+import java.io.ByteArrayOutputStream
+import java.nio.ByteBuffer
+
+
+/**
+ * Converts an [ImageProxy] to a [Bitmap].
+ * Currently only supports [ImageFormat.YUV_420_888] and [PixelFormat.RGBA_8888].
+ *
+ * @param applyRotation if true the resulting bitmap will be rotated to match target orientation of a use case.
+ * @throws [IllegalStateException] if the input format is [ImageFormat.YUV_420_888] and jpeg encoding of [YuvImage] fails.
+ * @see <a href="https://developer.android.com/reference/androidx/camera/core/ImageAnalysis.Builder#setOutputImageFormat(int)">ImageAnalysis supported output formats</a>
+ */
+public fun ImageProxy.toBitmap(applyRotation: Boolean = false): Bitmap {
+    // YUV_420_888 and RGBA_8888 are the only supported formats
+    val bitmap = when (format) {
+        ImageFormat.YUV_420_888 -> yuv4208888ToBitmap(this)
+        else -> rgba8888ToBitmap(this)
```

Comment:
Will the conversion work, if the format is unsupported? Perhaps we could throw an exception with a meaningful message here in this case.